core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -552,6 +552,12 @@ public void testToQuery() throws IOException {
         }
     }
 
```

Comment:
at the moment we test rewrite implicitly as we call it all the time right before calling toQuery. I think rewrite may deserve to be tested separately though? Just to make sure that it does what it's supposed to do. E.g. function score that rewrites its function filters too etc. otherwise we only test the input and the output but not the intermediate format. What do you think?